src/jit/importer.cpp

```diff
@@ -2207,6 +2207,13 @@ bool Compiler::impSpillStackEntry(unsigned level,
     /* Assign the spilled entry to the temp */
     impAssignTempGen(tnum, tree, verCurrentState.esStack[level].seTypeInfo.GetClassHandle(), level);
 
+    // If temp is single def and a ref type, grab what type info we can.
+    if (lvaTable[tnum].lvIsTemp && (lvaTable[tnum].lvType == TYP_REF))
+    {
+        CORINFO_CLASS_HANDLE stkHnd = verCurrentState.esStack[level].seTypeInfo.GetClassHandle();
+        lvaSetClass(tnum, tree, stkHnd);
+    }
```

Comment:
Right, I don't rely on `lvIsTemp` == "single def" to be true in general (otherwise I would do a more comprehensive type capture over in `impAssignTempGen`). But in this specific context the temps are single def when `lvIsTemp` is true.

Seems like we ought to fix up those places where `lvIsTemp` does not imply a single def with an eye towards ultimately making this reliable, at least through the importer / inliner.